### PR TITLE
Accessibility Issues Provider

### PR DESCRIPTION
**Pull request by tMaxx**
Implementation of #557 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] I have reviewed (and updated if needed) the documentation regarding this change

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: documentation/src/main/wiki/AccessibilityIssuesReport.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+#### Accessibility Issues Report
+
+This feature allows to generate a summary of various reported accessibility issues for a given suite.
+
+You can access it either through the web GUI, or query the report generation endpoint directly (for integration with build tools like Jenkins, Bamboo, ...).
+
+You can choose between two output file formats:
+- XLSX (Excel) - gives a summary of all issues by given type in a table format
+- TXT (pure text) - targetted as a helper for reporting these issues in JIRA (is already formatted for copy-paste)
```
Comment (tkaik) [inline documentation/src/main/wiki/AccessibilityIssuesReport.md:9]: Doesn't it mean that this text is simply formatted using Markdown?

Comment (tMaxx) [reply documentation/src/main/wiki/AccessibilityIssuesReport.md:9]: JIRA syntax is not exactly Markdown, so stating here that it is MD would be misleading

File: documentation/src/main/wiki/AccessibilityIssuesReport.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+After clicking on "Generate report", a new tab will open, and the report will start downloading to your computer. 
+
+###### Endpoints
+
+**Downloading the report**
+
+URL is as follows:
+
+`<domain>/api/accessibility/report?company=<company>&project=<project>&suite=<suite>&correlationId=<correlationId>&type=<type>&extension=<type>`
```
Comment (annabudziszewska) [inline documentation/src/main/wiki/AccessibilityIssuesReport.md:56]: 'type' should be replaced by 'verbosity'

File: documentation/src/main/wiki/AccessibilityIssuesReport.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+Where:
+- `domain`: domain on which AETs are set up (eg. `localhost`)
+- `company`, `project`, `suite`, `correlationId`: for these the report will be generated, if available
+- `verbosity`: verbosity level of the report, comma separated values: `NOTICE`, `WARNING`, `ERROR` (case-insensitive; e.g.: `notice,error`)
```
Comment (annabudziszewska) [inline documentation/src/main/wiki/AccessibilityIssuesReport.md:61]: `WARNING` is not working (it's 'WARN' instead) so documentation should be adjusted or implementation changed

**Comment by annabudziszewska** [comment]
1. Documentation should be adjusted:
documentation/src/main/wiki/AccessibilityIssuesReport.md
In section 'Downloading the report' in URL, 'type' should be replaced by 'verbosity'

2. Documentation is incoherent with the implementation
In report URL, WARNING is not working (as described in documentation), but 'WARN'
So this is working URL: http://localhost/api/accessibility/report?company=cognifide&project=pro&suite=sui&correlationId=corrID&verbosity=WARN&extension=txt
So this is NOT working URL: http://localhost/api/accessibility/report?company=cognifide&project=pro&suite=sui&correlationId=corrID&verbosity=WARNING&extension=txt

**Comment by annabudziszewska** [comment]
3. XLSX file generated with NOTICE verbosity (0 issues) , cannot be opened. The following error occurred
2020-10-30_15h50_15

4. IMPROVEMENT
It would be useful if the report would have an information with issue level (`NOTICE`, `WARNING`, `ERROR`)

**Pull request merged**